module/src/ranking_challenge/request.py

```diff
@@ -19,9 +20,11 @@ class TwitterEngagements(BaseModel):
 
 class RedditEngagements(BaseModel):
     """Engagement counts from Reddit"""
-
-    upvote: NonNegativeInt
-    downvote: NonNegativeInt
+    
+    warnings.warn('The upvotes and `downvotes` field is deprecated as it is net of upvotes and downvotes; use `score` instead.', DeprecationWarning)

```

Comment:
Will this warn whenever something imports RedditEngagements? If so, I don't think that's quite what we want.

I was looking to see if you could define custom getters/setters for this purpose, and how that would play with pydantic, and I stumbled across some docs suggesting that there's a way to deprecate fields built in already.

Docs are here: https://docs.pydantic.dev/latest/api/fields/#pydantic.fields.Field (search for deprecated). But the upshot is that I think you could do this:

```
upvote: Optional[NonNegativeInt] = Field('The number of upvotes (deprecated, use score instead)', deprecated='The reddit upvote field is deprecated, use score instead', default=None)
```

This also, I think, has the effect of making the field truly optional. Like it could be omitted completely from the json and will just get the default value. But you may also need to set the default to 0 for that to work, not sure. You can find out by trying to make one. Something like:

```
$ cd module/src
$ python
>>> from ranking_challenge.request import RedditEngagements
>>> RedditEngagements(score=4, comment=2, award=0)
```

If it doesn't raise an exception, then the field is actually optional. Currently it does raise an exception.

(or you could write a unit test that does the same thing)